WeakAuras/Prototypes.lua

```diff
@@ -11244,9 +11288,160 @@ Private.event_prototypes = {
+            local cloneActive = not primaryCheckFailed
+            local cloneId = currencyData.characterGUID or tostring(i)
+            states[cloneId] = states[cloneId] or {}
+            local s = states[cloneId]
+            s.value = currencyData.quantity
+            s.realCharacterQuantity = currencyData.realCharacterQuantity
+            s.characterName = currencyData.characterName
+            s.characterGUID = currencyData.characterGUID
+            s.mainCharacter = false
+
+            -- Check quantity-related values
+            if cloneActive and triggerQuantities.quantity ~= nil then
+              cloneActive = check(currencyData.quantity, triggerQuantities.quantity, triggerOperators.quantity)
+            end
+            if cloneActive and triggerQuantities.realCharacterQuantity ~= nil then
+              cloneActive = check(currencyData.realQuantity, triggerQuantities.realCharacterQuantity, triggerOperators.realCharacterQuantity)
+            end
+            -- No warband currencies exist with a season or weekly max, hence we only need to check for a maxQuantity
+            if cloneActive and capped ~= "nil" and currencyInfo.maxQuantity and currencyInfo.maxQuantity > 0 then
+              cloneActive = tostring((currencyData.quantity or 0) >= (currencyInfo.maxQuantity or 0)) == capped
+            end
+
+            s.show = cloneActive
+          end
+        end
+
+        -- Apply shared values to states
+        for _, state in pairs(states) do
+          for key, value in pairs(sharedStateValues) do
+            state[key] = value
+          end
+        end
+
+        return true
+      end
+      ]=]
+
+      return ret:format(
+        trigger.currencyId or 1,
+        trigger.use_value and tonumber(trigger.value) or "nil",
+        trigger.use_realCharacterQuantity and tonumber(trigger.realCharacterQuantity) or "nil",
+        trigger.use_accountQuantity and tonumber(trigger.accountQuantity) or "nil",
+        trigger.use_realAccountQuantity and tonumber(trigger.realAccountQuantity) or "nil",
+        trigger.use_maxQuantity and tonumber(trigger.maxQuantity) or "nil",
+        trigger.use_quantityEarnedThisWeek and tonumber(trigger.quantityEarnedThisWeek) or "nil",
+        trigger.use_totalEarned and tonumber(trigger.totalEarned) or "nil",
+        trigger.use_value and trigger.value_operator or "<",
+        trigger.use_realCharacterQuantity and trigger.realCharacterQuantity_operator or "<",
+        trigger.use_accountQuantity and trigger.accountQuantity_operator or "<",
+        trigger.use_realAccountQuantity and trigger.realAccountQuantity_operator or "<",
+        trigger.use_maxQuantity and trigger.maxQuantity_operator or "<",
+        trigger.use_quantityEarnedThisWeek and trigger.quantityEarnedThisWeek_operator or "<",
+        trigger.use_totalEarned and trigger.totalEarned_operator or "<",
+        trigger.use_discovered ~= nil and (trigger.use_discovered and "true" or "false") or "nil",
```

Comment:
Using something like "trigger.use_discovered and "true" or "false" " is fine, if the check is not set at all, we can treat it as "false".

Reply:
Actually you are right here, this is a tristate value, which is the stupid kind of unreadable code we sometimes create.